book/src/development/define_lints.md

```diff
@@ -0,0 +1,189 @@
+# Define New Lints
+
+The first step in the journey of a new lint is to define the definition
+and registration of the lint in Clippy's codebase.
+We can use the Clippy dev tools to handle this step since setting up the 
+lint involves some boilerplate code.
+
+## Name the Lint
+
+A wise software engineer Phil Karlton once said:
+> There are only two hard things in Computer Science: cache invalidation and naming things.
+
+Naming a lint is no exception.
+Therefore, in case of uncertainty about if the name you chose fits the lint,
+please do the following:
+
+1. Consult our [lint naming guidelines][lint_naming]
+2. Ping a [Clippy team member][clippy_team_members] in our [Zulip] chat
+3. Comment on the corresponding Github issue (less preferrable as comments might be overlooked)
+
+For now, let's imagine that your fellow developers at work tend to define some of their
+functions with the name `foo` and forget to re-name it to a more meaningful name
+when they submit a pull request.
+
+`foo` is a highly non-descriptive name for a function, so we want to detect this
+bad naming and fix it early on in the development process.
+
+For this, we will create a lint that detects these `foo` functions and
+help our fellow developers correct this bad practice. Note that in Clippy,
+lints are generally written in snake cases.
+We can name this new lint `foo_functions`.
```

Comment:
This is for example completely unnecessary in this section. Those subsections shouldn't contain story telling, but rather hard and cold facts about what has to be done to define a lint.

The story telling is nice for reading this documentation and has its place. But I think its place is what currently is the adding-lints section that goes through the process. I would still keep this chapter and include this story telling and creative writing. But those sub-sections should only be instructions on how to do things.

---

So this can be collapsed to

```
### Example

A lint that checks function names and will trigger on all functions named `foo` would be named `foo_functions`. 

/// Maybe add negative examples like `foo_function` here and explain why this name would not be chosen
```